src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -281,66 +281,68 @@ private static ChatRole ToChatRole(global::Azure.AI.Inference.ChatRole role) =>
     private ChatCompletionsOptions ToAzureAIOptions(IEnumerable<ChatMessage> chatContents, ChatOptions? options)
     {
-        ChatCompletionsOptions result = new(ToAzureAIInferenceChatMessages(chatContents))
+        if (options is null)
         {
-            Model = options?.ModelId ?? _metadata.DefaultModelId ?? throw new InvalidOperationException("No model id was provided when either constructing the client or in the chat options.")
-        };
+            return CreateAzureAIOptions(chatContents, options);
+        }
 
-        if (options is not null)
+        if (options.RawRepresentationFactory?.Invoke(this) is ChatCompletionsOptions result)
+        {
+            result.Messages = ToAzureAIInferenceChatMessages(chatContents).ToList();
+            result.Model ??= options.ModelId ?? _metadata.DefaultModelId ??
+                throw new InvalidOperationException("No model id was provided when either constructing the client or in the chat options.");
+        }
+        else
         {
-            result.FrequencyPenalty = options.FrequencyPenalty;
-            result.MaxTokens = options.MaxOutputTokens;
-            result.NucleusSamplingFactor = options.TopP;
-            result.PresencePenalty = options.PresencePenalty;
-            result.Temperature = options.Temperature;
-            result.Seed = options.Seed;
-
-            if (options.StopSequences is { Count: > 0 } stopSequences)
+            result = CreateAzureAIOptions(chatContents, options);
+        }
+
+        result.FrequencyPenalty ??= options.FrequencyPenalty;
+        result.MaxTokens ??= options.MaxOutputTokens;
+        result.NucleusSamplingFactor ??= options.TopP;
+        result.PresencePenalty ??= options.PresencePenalty;
+        result.Temperature ??= options.Temperature;
+        result.Seed ??= options.Seed;
+
+        if (options.StopSequences is { Count: > 0 } stopSequences)
+        {
+            foreach (string stopSequence in stopSequences)
             {
-                foreach (string stopSequence in stopSequences)
-                {
-                    result.StopSequences.Add(stopSequence);
-                }
+                result.StopSequences.Add(stopSequence);
             }
+        }
 
-            // These properties are strongly typed on ChatOptions but not on ChatCompletionsOptions.
-            if (options.TopK is int topK)
+        if (options.AdditionalProperties is { } props)
+        {
+            foreach (var prop in props)
             {
-                result.AdditionalProperties["top_k"] = new BinaryData(JsonSerializer.SerializeToUtf8Bytes(topK, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(int))));
+                byte[] data = JsonSerializer.SerializeToUtf8Bytes(prop.Value, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(object)));
+                result.AdditionalProperties[prop.Key] = new BinaryData(data);
             }
```

Comment:
We still need this one, don't we? This is ensuring that the strongly-typed property on ChatOptions is respected if the underlying provider supports it.

Reply:
Yes, but we should write it only if it's not already set in the additional properties of raw representation, correct?

Reply:
That sounds like a good idea.